component/src/main/java/io/siddhi/extension/store/mongodb/MongoSelectExpressionVisitor.java

```diff
@@ -0,0 +1,307 @@
+        if (isCountFunction) {
+            throw new MongoTableException("The MongoDB Event table does not support arguments for count function.");
+        }
+        conditionOperands.push("\'$" + attributeName + "\'");
+    }
+
+    @Override
+    public void endVisitStoreVariable(String storeId, String attributeName, Attribute.Type type) {
+    }
+
+    @Override
+    public void beginVisitStreamVariable(String id, String streamId, String attributeName, Attribute.Type type) {
+        if (isCountFunction) {
+            throw new MongoTableException("The MongoDB Event table does not support functions arguments for count " +
+                    "function.");
+        }
+        if (isNullCheck) {
+            throw new MongoTableException("The MongoDB Event table does not support null check for stream variables.");
+        }
+        if (type.toString().equalsIgnoreCase("STRING")) {
+            conditionOperands.push("{\'$literal\':\'\'" + id + "\'\'}");
+        } else {
+            conditionOperands.push("{\'$literal\':\'" + id + "\'}");
+        }
+    }
+
+    @Override
+    public void endVisitStreamVariable(String id, String streamId, String attributeName, Attribute.Type type) {
+    }
+
+    @Override
+    public void beginVisitAttributeFunction(String namespace, String functionName) {
+        if (!MongoTableUtils.isEmpty(namespace) ||
+                (!Arrays.asList(supportedFunctions).contains(functionName))) {
+            throw new MongoTableException("The MongoDB Event table does not support functions other than sum(), " +
+                    "avg(), min(), max() and count().");
+        }
+        if (functionName.equalsIgnoreCase("count")) {
+            this.isCountFunction = true;
+        }
+    }
+
+    @Override
+    public void endVisitAttributeFunction(String namespace, String functionName) {
+        if (MongoTableUtils.isEmpty(namespace) &&
+                (Arrays.asList(supportedFunctions).contains(functionName))) {
+            this.isAttributeFunctionUsed = true;
+            if (functionName.equalsIgnoreCase("count")) {
+                this.isCountFunction = false;
+                conditionOperands.push("{$sum:1}");
+            } else {
+                String functionArgument = conditionOperands.pop();
+                conditionOperands.push("{$" + functionName + ":" + functionArgument + "}");
+            }
+        }
+    }
+
+    @Override
+    public void beginVisitParameterAttributeFunction(int index) {
+    }
+
+    @Override
+    public void endVisitParameterAttributeFunction(int index) {
+    }
+
+    @Override
+    public void beginVisitMath(MathOperator mathOperator) {
+    }
+
+    @Override
+    public void endVisitMath(MathOperator mathOperator) {
+        String rightOperand = this.conditionOperands.pop();
+        String leftOperand = this.conditionOperands.pop();
+        conditionOperands.push("{$" + mathOperator.name().toLowerCase() + ":" + "[" + leftOperand + "," +
```

Comment:
Use constants similar to https://github.com/siddhi-io/siddhi-store-mongodb/blob/master/component/src/main/java/io/siddhi/extension/store/mongodb/util/MongoTableConstants.java#L51

Reply:
Fixed in 
https://github.com/siddhi-io/siddhi-store-mongodb/pull/56/commits/94fba768238670966474ed6ac8a757455790895b